packages/canvas-panel-core/src/components/ThumbnailList/ThumbnailList.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withBemClass } from '@canvas-panel/core';
+//import './ThumbnailList.scss';
+
+const thumbnailGetSize = (thumbnail, pWidth, pHeight) => {
+  const thumb = thumbnail.__jsonld;
+  if (
+    (pWidth || pHeight) &&
+    thumb.hasOwnProperty('service') &&
+    thumb.service.hasOwnProperty('sizes')
+  ) {
+    let closestSizeIndex = -1;
+    let minDistanceX = Number.MAX_SAFE_INTEGER;
+    let minDistanceY = Number.MAX_SAFE_INTEGER;
+    thumb.service.sizes.forEach((size, index) => {
+      if (pWidth) {
+        const xDistance = Math.abs(size.width - pWidth);
+        if (minDistanceX >= xDistance) {
+          closestSizeIndex = index;
+          minDistanceX = xDistance;
+        }
+      }
+      if (pHeight) {
+        const yDistance = Math.abs(size.height - pHeight);
+        if (minDistanceY >= yDistance) {
+          closestSizeIndex = index;
+          minDistanceY = yDistance;
+        }
+      }
+    });
+    let thumbUrlParts = (thumb.id || thumb['@id']).split('/');
+    if (closestSizeIndex !== -1) {
+      const size = thumb.service.sizes[closestSizeIndex];
+      thumbUrlParts[thumbUrlParts.length - 3] = [size.width, size.height].join(
+        ','
+      );
+    }
+    return thumbUrlParts.join('/');
+  } else {
+    return thumb.id || thumb['@id'];
+  }
+};
+
+class ThumbnailList extends React.Component {
+  thumbnailCache = {};
+
+  componentDidUpdate(/*prevProps, prevState*/) {
+    if (this.selectedThumbnail) {
+      const list = this.selectedThumbnail.parentNode.parentNode;
+      const rect = this.selectedThumbnail.getBoundingClientRect();
+      if (rect.x < 0) {
+        list.scrollLeft = 0;
+      } else if (list.offsetWidth - rect.width < rect.x) {
```

Comment:
Am I right in saying that if we switched  `scrollLeft` to `scrollTop`, `offsetWidth` to `offsetTop`, `rect.width` to `rect.height` and `rect.x` to `rect.y` that it would also potentially work as a vertical thumbnail view, more like UV / mirador? Could be something we add in the future behind a prop

Reply:
> TODO: maybe vertical list
Just saw that in the description, cool!

Reply:
added default styling, vertical and multi-column. https://github.com/digirati-co-uk/canvas-panel/pull/370/commits/ed3f9bccb5da0d161eea2a66df5e512f15438143